src/target_info.rs

```diff
@@ -1,29 +1,39 @@
 use object::{Object, ObjectSection as _};
 use probe_rs::config::{MemoryRegion, RamRegion};
-use std::convert::TryInto;
+use std::{convert::TryInto, ops::RangeInclusive};
 
 use crate::elf::Elf;
 
 pub(crate) struct TargetInfo {
     pub(crate) probe_target: probe_rs::Target,
     /// RAM region that contains the call stack
     pub(crate) active_ram_region: Option<RamRegion>,
-    /// Only `Some` if static variables are located in the `active_ram_region`
-    pub(crate) highest_static_var_address: Option<u32>,
+    /// Valid values of the stack pointer (that don't collide with other data).
+    pub(crate) stack_range: Option<RangeInclusive<u32>>,
+    pub(crate) data_below_stack: bool,
 }
 
 impl TargetInfo {
     pub(crate) fn new(chip: &str, elf: &Elf) -> anyhow::Result<Self> {
         let probe_target = probe_rs::config::get_target_by_name(chip)?;
         let active_ram_region =
             extract_active_ram_region(&probe_target, elf.vector_table.initial_stack_pointer);
-        let highest_static_var_address =
-            extract_highest_static_var_address(elf, active_ram_region.as_ref());
+        let stack_range = extract_stack_range(
+            elf,
+            active_ram_region.as_ref(),
+            elf.vector_table.initial_stack_pointer,
+        );
+
+        let data_below_stack = match (&active_ram_region, &stack_range) {
```

Comment:
nit: this looks to me that the `TargetInfo` field should have type `Option<StackInfo>` and the `StackInfo` struct should be `{ stack_range: RangeInclusive<u32>, data_below_stack: bool }`